scripts/download-import-schema-from-gecko-dev

```diff
@@ -0,0 +1,130 @@
+  paths: [
+    'browser/config/version.txt',
+    'browser/config/version_display.txt',
+    'toolkit/components/extensions/schemas',
+    'browser/components/extensions/schemas',
+    'mobile/android/components/extensions/schemas',
+  ],
+  partialCloneOptions: '--depth 1 --filter=blob:none --no-checkout',
+  tmpDir: path.join('tmp', `gecko-dev--partial-${Date.now()}`),
+};
+
+function processScriptOptions() {
+  const customBranch = process.argv[2];
+  if (customBranch) {
+    if (!['master', 'beta'].includes(customBranch)) {
+      shell.echo(`ERROR: Invalid gecko-dev branch name "${customBranch}"`);
+      return false;
+    }
+    gitClone.branch = customBranch;
+  }
+  return true;
+}
+
+function checkPendingSchemaChanges() {
+  const oldFatal = shell.config.fatal;
+  shell.config.fatal = false;
+  const result = shell.exec(`git diff --exit-code src/schema/imported`, {
+    silent: true,
+  });
+  if (result.code === 1) {
+    shell.echo(
+      'ERROR: Unexpected pending changes in src/schema/imported.\n\n' +
+        'Existing changes should be committed or cleared before importing new schema data:\n'
+    );
+    shell.echo(result.stdout);
+    return false;
+  }
+  shell.config.fatal = oldFatal;
+  return true;
+}
+
+function partialClone() {
+  const { branch, partialCloneOptions, repo, tmpDir, paths } = gitClone;
+
+  shell.echo(`Partial Cloning gecko-dev brach ${branch}`);
+
+  if (shell.test('-e', tmpDir)) {
+    throw new Error(
+      `Temporary directory "${tmpDir}" already exists. Remove it before re-running this script`
+    );
+  }
+
+  shell.mkdir('-p', tmpDir);
+
+  // Creates the partial clone dir.
+  shell.exec(`git clone -b ${branch} ${partialCloneOptions} ${repo} ${tmpDir}`);
+  // Checkout the working tree paths we need for the import.
+  shell.exec(`git checkout ${branch} -- ${paths.join(' ')}`, {
+    silent: true,
+    cwd: tmpDir,
+  });
+
+  shell.echo(`Partial clone completed\n`);
+}
+
+function importSchemaFromPartialClone() {
+  const { tmpDir } = gitClone;
+
+  const version_file_path = path.join(
+    tmpDir,
+    'browser/config/version_display.txt'
+  );
+  if (!shell.test('-e', version_file_path)) {
+    throw new Error(
+      `"${version_file_path}" does not exist. Partial gecko-dev git clone failed`
+    );
+  }
+
+  const version_display = shell.cat(version_file_path);
+  shell.echo(
+    `Importing WebExtensions API JSONSchema data from Gecko ${version_display}`
+  );
+
+  shell.exec(`./scripts/firefox-schema-import ${tmpDir}`);
+
+  shell.echo('Schema changes diff:\n');
+
+  const oldFatal = shell.config.fatal;
+  shell.config.fatal = false;
+  if (shell.exec(`git diff --exit-code src/schema/imported`).code === 0) {
+    shell.echo('No changes to the schema data');
+  }
+  shell.config.fatal = oldFatal;
+}
+
+function cleanTmpDir() {
+  const { tmpDir } = gitClone;
+  // Remove the tempDir by default, unless environment variable is set.
+  if (process.env.KEEP_GECKODEV_TMPDIR == null && tmpDir) {
```

Comment:
Why do we need such an env var? (also this env var should be checked against `'1'` or something because I'd expect `KEEP_GECKODEV_TMPDIR=0` to skip this block)

Reply:
by default I prefer that this removes the git clone, but for debugging purpose would be useful to be able to turn off that and be able to see the partial clone content and be able to run the import script manually for further investigations.

Given that it is just for debugging purpose I preferred to allow it through an environment variable instead of a command line option.

I'm definitely ok on doing a more strict check on the value set in the environment variables.